zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/factory/SharedZkClientFactory.java

```diff
@@ -82,13 +94,8 @@ public HelixZkClient buildZkClient(HelixZkClient.ZkConnectionConfig connectionCo
         throw new ZkClientException("Failed to create a connection manager in the pool to share.");
       }
       LOG.info("Sharing ZkConnection {} to a new SharedZkClient.", connectionConfig.toString());
-      return new SharedZkClient(zkConnectionManager, clientConfig,
-          new SharedZkClient.OnCloseCallback() {
-            @Override
-            public void onClose() {
-              cleanupConnectionManager(zkConnectionManager);
-            }
-          });
+      return new InnerSharedZkClient(zkConnectionManager, clientConfig,
```

Comment:
nit: the intention is to keeper old sharedZkClient the same, but use new name innerSharedZkClient.  let us don't use the new Java feature to rewrite the same logic. Just leave it as what it were. It would be easier to people to look at the history to understand the intention.